Revamp EKS Fargate documentation

JacksonDavenport opened the pull request:
### What does this PR do?
<!-- A brief description of the change being made with this pull request. -->
[TEEP-601](https://datadoghq.atlassian.net/browse/TEEP-601)

This PR grew a bit more than expected and is effectively a big rewrite of this doc.
- Moves a lot of the instructions to prerequisite steps, especially as those were in common between Operator/Helm/Manual
- Centralize and clarify a lot of the contextual information at the top
- Clarify use of Secret for API Keys and Tokens, all manual setups use the Secret ref instead of raw API Key
- Overhaul how we recommend to do the RBAC to
  - Clarify/stress how this is used in the pods
  - Provide examples on how to configure it per namespace
  - Provide examples on how to configure it for multiple pre-existing service accounts
  - Avoid name clashes with the Helm/Operator installs (ClusterRole/ClusterRoleBinding: `datadog-agent` -> `datadog-agent-fargate`)
  - Provide a troubleshooting `kubectl auth` command to validate
- General cleanup on Operator/Helm install methods, especially for samples, results, and header structure
- General cleanup on manual install methods
- Update integrations to Autodiscovery V2 syntax
- Fix some inaccuracies for what is/isn't collected
- Fix a lot of broken and orphaned links

Preview Link:
- https://docs-staging.datadoghq.com/cswatt/eks-preview/integrations/eks_fargate/?tab=admissioncontrollerhelm#trace-collection

### Motivation
<!-- What inspired you to submit this pull request? -->
An accumulation of different customer and TSE experiences with this doc. Originally was just for RBAC and Keys, but turned into a full revamp.

Did NOT cover how APM Auto Instrumentation plays with EKS Fargate. That is a whole other topic that is probably worth a separate guide page to tackle compared to this.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Add the `qa/skip-qa` label if the PR doesn't need to be tested during QA.
- [ ] If you need to backport this PR to another branch, you can add the `backport/<branch-name>` label to the PR and it will automatically open a backport PR once this one is merged


[TEEP-601]: https://datadoghq.atlassian.net/browse/TEEP-601?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

gabedos commented on eks_fargate/README.md:
```diff
@@ -5,61 +5,59 @@
-**Note**: Cloud Network Monitoring (CNM) is not supported for EKS Fargate.
-
-## Setup
+EKS Fargate pods not run on traditional EKS nodes and EC2 instances. While the Agent does report [system checks][1], like `system.cpu.*` and `system.memory.*`, these are just for the Agent container. In order to collect data from your AWS Fargate pods, run the Agent as a sidecar within *each* of your desired application pods. The pod needs to be deployed with custom RBAC granting permissions to t ...
```
Nit: typo and maybe improved wording:
Prev: `EKS Fargate pods not run on traditional EKS nodes and EC2 instances.`
New: `EKS Fargate pods do not run on traditional EKS nodes backed by EC2 instances.`

domalessi commented:
Created DOCS-10710 to track documentation team review

JacksonDavenport commented on eks_fargate/README.md:
```diff
@@ -156,562 +188,643 @@ The setup below configures the Cluster Agent to communicate with the Agent sidec
-**Notes**:
-
-- Don't forget to replace `<YOUR_DATADOG_API_KEY>` with the [Datadog API key from your organization][14].
-- Container metrics are not available in Fargate because the `cgroups` volume from the host can't be mounted into the Agent. The [Live Containers][17] view reports 0 for CPU and Memory.
```
I think that this line is just wrong, I can get standard `kubernetes.cpu` and `kubernetes.memory` metrics, generic `container.cpu` and `container.memory` metrics, as well as Live Containers by default. Then Live Process metrics once enabled. 

That looks like a note we may have added way back when we first released EKS Fargate support and before we just started going to the kubelet for everything here. https://github.com/DataDog/integrations-core/pull/6045

gabedos commented on eks_fargate/README.md:
````diff
@@ -156,562 +188,643 @@ The setup below configures the Cluster Agent to communicate with the Agent sidec
+          # (...)
+```
 
 ## Log collection
````
Future note:

Soon we'll have logging directly via the Agent https://github.com/DataDog/datadog-agent/pull/35190

JacksonDavenport replied:
Oh that's really good, EKS Fargate log collection has been a bit of a pain point. *This* PR itself doesn't really change the logs setup. So a future one for APM/Auto-Instrumentation (probably a separate guide page) would be good. As well as one to add these instructions to this doc, and IMO move this strategy to a separate page to keep this one trim - assuming that Agent based collection is just all around superior.

JacksonDavenport commented on eks_fargate/README.md:
```diff
@@ -5,61 +5,59 @@
-- [Autodiscovery][4]
-- Configuration of custom Agent Checks to target containers in the same pod
-- APM and DogStatsD for containers in the same pod
+If you have a mixed cluster of traditional EKS Nodes and Fargate pods, you will manage the traditional Datadog Kubernetes installation with a Helm Chart or Datadog Operator, and then manage your EKS Fargate pods separately.
```
Added that extra confirmation there

JacksonDavenport commented on eks_fargate/README.md:
````diff
@@ -156,562 +188,643 @@ The setup below configures the Cluster Agent to communicate with the Agent sidec
 ```
 
-**Note**: Don't forget to replace `<YOUR_DATADOG_API_KEY>` with the [Datadog API key from your organization][14].
+**Note**: Add `DD_TAGS` to append additional space separated `<KEY>:<VALUE>` tags. The `DD_CLUSTER_NAME` environment variable will set your `kube_cluster_name` tag.
````
I think this should be fine, I haven't seen anyone use anything in the 7.3X in a while. That was released March 2022 so everyone should hopefully be above this. Especially those setting up new EKS Fargate environments.

JacksonDavenport commented on eks_fargate/README.md:
```diff
@@ -156,562 +188,643 @@ The setup below configures the Cluster Agent to communicate with the Agent sidec
+### Kubernetes resources view
 
-Datadog Agent v6.19+ supports live processes in the EKS Fargate integration. Live processes appear on the [Processes][20] page. To enable live processes, [enable shareProcessNamespace in the pod spec][21].
+To collect Kubernetes resource views, you need a [Cluster Agent setup][17] and a valid connection between the sidecar Agent and Cluster Agent. This is connected for you automatically if you are using the sidecar injection setup.
```
Added some clarification 👍 

> To collect Kubernetes resource views, you need a [Cluster Agent setup][17] and a valid connection between the sidecar Agent and Cluster Agent. When using the Admission Controller's sidecar injection setup this is connected for you automatically. When configuring the sidecar manually ensure you are [connecting the sidecar Agent](#configuring-sidecar-for-cluster-agent).

JacksonDavenport commented:
@cswatt Ready for your re-review on this. Added the config snippets for Process Collection, also simplified the APM/DogStatsD portion. TLDR: You don't _really_ need to specify a port like this for connectivity. It can be a nice to have for clarity sake and when using K8s services. 
```yaml
ports:
  - name: traceport
    containerPort: 8126
    protocol: TCP
```
However connectivity is totally fine without it. Additionally we do not currently support adding ports with the Admission Controller so didn't want users to think they are stuck without it if not using the "Manual" approach.

cswatt commented:
Updated staging preview link: https://docs-staging.datadoghq.com/cswatt/eks-preview/integrations/eks_fargate
